Entity/OverrideLeadFieldRepository.php

```diff
@@ -212,29 +221,46 @@ public function compareValue($lead, $field, $value, $operatorExpr)
     }
 
     /**
-     * Get an extended field given the field alias.
-     *
-     * @param string $alias
+     * @param $fieldAlias
      *
-     * @return null|array
+     * @return array|bool|mixed|null
      */
-    private function getExtendedField($alias)
+    private function getExtendedFieldConfigurations($fieldAlias)
     {
-        $qf = $this->_em->getConnection()->createQueryBuilder();
-        $qf->select('lf.id, lf.object, lf.type, lf.alias, lf.field_group as "group", lf.object, lf.label')
-            ->from(MAUTIC_TABLE_PREFIX.'lead_fields', 'lf')
-            ->where(
-                $qf->expr()->andX(
-                    $qf->expr()->eq('lf.alias', ':alias'),
-                    $qf->expr()->orX(
-                        $qf->expr()->eq('lf.object', $qf->expr()->literal('extendedField')),
-                        $qf->expr()->eq('lf.object', $qf->expr()->literal('extendedFieldSecure'))
-                    )
-                )
-            )
-            ->setParameter('alias', $alias);
+        if (!$this->extendedFieldConfigurations) {
+            $cacheHelper     = new CacheStorageHelper(
+                CacheStorageHelper::ADAPTOR_FILESYSTEM,
+                'extendedFieldConfiguration',
+                null,
+                $this->coreParametersHelper->getParameter('mautic.cache_path'),
+                60
+            );
+            if (empty($extendedFieldConfigurations = $cacheHelper->get('extendedFieldConfiguration', 60))) {
```

Comment:
I've seen this before, but I don't really understand it.
`empty($extendedFieldConfigurations = $cacheHelper->get('extendedFieldConfiguration', 60))`
It's setting `$extendedFieldConfigurations` to the cached version, and then evaluating if $extendedFieldConfigurations is not null?

Reply:
Yes, that's what he's doing here.
```
php > var_dump(boolval(empty($a = null)));
bool(true)
php > var_dump(boolval(empty($a = 1)));
bool(false)
```

The `empty` in this case isn't technically required, but scrutinizer might demand it.

Reply:
not necessarily null, could be an empty array. Not likely in a system that actually has fields. but technically, cache is set to be an array of custom field configs, and no custom fields would be an empty array. if there is no instance of the cache key, then null may be returned.